content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -1,8 +1,162 @@
+> `dvc repro` and `dvc commit` commands, but not when a `.dvc` file is
+> overwritten by `dvc add`,`dvc import`, or `dvc import-url`.
+
+## dvc.yaml files
+
+When you add commands to a pipeline with `dvc run`, the `dvc.yaml` file is
+created or updated. Here's a simple example:
+
+```yaml
+stages:
+  stageone:
+    cmd: python cmd.py input.data output.data metrics.json
+    deps:
+      - cmd.py
+      - input.data
+    outs:
+      - output.data
+    metrics:
+      - metrics.json
+  stagetwo:
+    cmd: python ...
+    meta: '2nd stage' # User metadata and comments are supported.
+    deps: ...
+```
+
+`dvc.yaml` files consists of a group of `stages` with names provided explicitly
+by the user with the `--name` (`-n`) option of `dvc run`.
+
+Each stage's contents are similar to individual [`dvc` files](#dvcfiles) but
+they can contain more information in `dvc.yaml` These are the possible following
+fields:
+
+- `cmd`: Executable command defined in this stage
+- `deps` (optional): List of <abbr>dependencies</abbr> for this stage
+- `outs` (optional): List of <abbr>outputs</abbr> for this stage
+- `params` (optional): List of the [parameter](/doc/command-reference/params)
+  names and their current values
+- `metrics` (optional): List of [metrics](/doc/command-reference/metrics)
+- `plots` (optional): List of [plot metrics](/doc/command-reference/plots)
+- `frozen` (optional): Whether or not this stage is frozen from reproduction
+- `always_changed` (optional): Whether or not this stage is considered as
+  changed by commands such as `dvc status` and `dvc repro`. `false` by default
+- `meta` (manual): Arbitrary information can be added here manually. Any YAML
+  contents can be added. `meta` contents are ignored by DVC, but they can be
+  useful for user processes that read `.dvc` files.
+
+An output entry consists of these fields:
+
+- `md5`: Hash value for the output file
+- `path`: Path to the output in the <abbr>workspace</abbr>, relative to the
+  location of the `.dvc` file
+- `cache` (optional): Whether or not DVC should cache the output. `true` by
+  default
+
+A `dvc.yaml` dependency entry consists of a these possible fields:
+
+- `path`: Path to the dependency, relative to the `wdir` path (always present)
+- `md5`: MD5 hash for the dependency (most [stages](/doc/command-reference/run))
+- `etag` (optional): Strong ETag response header (only HTTP <abbr>external
+  dependencies</abbr> created with `dvc import-url`)
+
+  > See the examples in
+  > [External Dependencies](/doc/user-guide/external-dependencies) for more
+  > info.
+
+Metrics entries can contain these fields:
+
+- `type`: Type of the metrics file (`json`)
+- `xpath`: Path within the metric file to the metrics data(e.g. `AUC.value` for
+  `{"AUC": {"value": 0.624321}}`)
+
+`dvc.yaml` files also support `# comments`.
+
+> `meta` fields and `#` comments are preserved among executions of `dvc run`,
+> `dvc repro`, and `dvc commit`.
```

Comment:
They should be always preserved. Unless user does `dvc remove stage-name` of-course. So, I think, it's redundant to even have a note.

Reply:
For .dvc files these are not preserved when running add again (I think), so we have a similar note for that section. So if you read that you may wonder about the same field in dvc.yaml, which is why we also have the note here. But maybe I can just say "they're always preserved".

Reply:
Updated in 910e6dbe